### ISSUE TITLE
Retaining last activated scene throws exception

### ISSUE THREAD

**Issue by koenekelschot** [azogue/eventsensor#22]
Using the provided example to retain the last activated scene homeassistant throws an exception when the sensor is triggered:

> 2020-12-13 17:26:20 ERROR (MainThread) [homeassistant] Error doing job: Exception in callback EventSensor.async_added_to_hass.<locals>.async_update_sensor(<Event call_s...n_videocall']>) at /config/custom_components/eventsensor/sensor.py:154
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/asyncio/events.py", line 81, in _run
    self._context.run(self._callback, *self._args)
  File "/config/custom_components/eventsensor/sensor.py", line 162, in async_update_sensor
    if new_state in self._state_map:
TypeError: unhashable type: 'list'
2020-12-13 17:26:20 ERROR (MainThread) [homeassistant] Error doing job: Exception in callback EventSensor.async_added_to_hass.<locals>.async_update_sensor(<Event call_s...n_videocall']>) at /config/custom_components/eventsensor/sensor.py:154
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/asyncio/events.py", line 81, in _run
    self._context.run(self._callback, *self._args)
  File "/config/custom_components/eventsensor/sensor.py", line 162, in async_update_sensor
    if new_state in self._state_map:
TypeError: unhashable type: 'list'

This happens with or without the statemap attribute. I'm using version 0.117.6 of homeassistant.
Is this a known issue?

**Comment by klatka** [comment]
This because the entity_id is a list of possible targets. The example assumes that it is only one entity_id.

If you listen to this event in dev tools you will get a scene.turn_on like this:

```
    "event_type": "call_service",
    "data": {
        "domain": "scene",
        "service": "turn_on",
        "service_data": {
            "entity_id": [
                "scene.x1"
            ]
        }
    }
```

**Comment by azogue** [comment]
Hi @koenekelschot, @klatka,

I've just published a new version ([v2.4.2](https://github.com/azogue/eventsensor/releases/tag/v2.4.2)) _trying_ to handle this issue, fix is in #34. 

As HA Core would **have trouble** with a state value consisting in  a complex object, like a collection (list or dict), the _EventSensor_ tries to honor simple types like **ints or floats** (to build numeric sensors!), **or expects a simple string as a state**.

It was doing a simple `str(state)` when the extracted state was a dict (producing ugly string states like `"{'entity_id': ['scene.x1']}"` for `service_data`, e.g.), **BUT** the case of a list was not handled (my fault, it was silly). 

Now it flattens the complex object into a string, producing states like `scene.x1` in the example above from @klatka, or `scene.x1,scene.x2` if the list has 2 scenes, using `service_data.entity_id` as state key.

I do not plan to go further in this _state parsing_ thing, so things like array access (`service_data.entity_id[1]`) are not in the roadmap, sorry :)
I'm totally open for collaborations, so if anyone wants to go that way, please do!

As the exception error should be fixed with #34 , I'm closing this issue ✅
